### PR TITLE
[gsoap] Add supports for gsoap

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**

- What does your PR fix? 
  Gsoap only supports `Win32`. Add `Supports` and `vcpkg_fail_port_install` for gsoap.
- No features need to test.

### CODE REVIEW

File: ports/gsoap/CONTROL [hunk 1]
```diff
@@ -2,4 +2,5 @@ Source: gsoap
 Version: 2.8.93-2
```
Comment (JackBoosY) [inline ports/gsoap/CONTROL:2]: Update version.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gsoap/portfile.cmake [hunk 2]
```diff
@@ -16,7 +16,6 @@ vcpkg_extract_source_archive_ex(
 if (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
     set(BUILD_ARCH "Win32")
```
Comment (strega-nil) [inline ports/gsoap/portfile.cmake:17]: Given that we should never get to this point, this should probably just not `set`.

File: ports/gsoap/CONTROL [hunk 3]
```diff
@@ -1,5 +1,6 @@
 Source: gsoap
-Version: 2.8.93-2
+Version: 2.8.93-3
 Build-Depends: curl
 Homepage: https://sourceforge.net/projects/gsoap2/
-Description: The gSOAP toolkit is a C and C++ software development toolkit for SOAP and REST XML Web services and generic C/C++ XML data bindings.
\ No newline at end of file
+Description: The gSOAP toolkit is a C and C++ software development toolkit for SOAP and REST XML Web services and generic C/C++ XML data bindings.
+Supports: x86
```
Comment (strega-nil) [inline ports/gsoap/CONTROL:6]: this should probably be `Supports: x86 & windows`

File: ports/gsoap/portfile.cmake [hunk 4]
```diff
@@ -16,7 +16,6 @@ vcpkg_extract_source_archive_ex(
 if (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
```
Comment (strega-nil) [inline ports/gsoap/portfile.cmake:16]: I don't understand why we don't remove this `if-else` now, since we don't support non-x86 non-windows.

**Comment by strega-nil** [comment]
Could you rerun after running the correct tombstone scripts?

**Comment by strega-nil** [comment]
LGTM! Thanks @LilyWangL :D

**Pull request merged**